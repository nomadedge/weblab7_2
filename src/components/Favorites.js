import React, { Component } from 'react';
import { connect } from 'react-redux';

import { addCity, deleteCity } from '../actions/FavoriteCitiesActions';

import FavoriteCityWeather from './FavoriteCityWeather';

class Favorites extends Component {
    handleAdd = event => {
        event.preventDefault();
        this.props.addCity(event.target[0].value, this.props.favoriteCities);
        event.target[0].value = '';
    }

    handleDelete = cityName => {
        this.props.deleteCity(cityName);
    };

    render() {
        return (
            <>
                <form name='cityAdd' onSubmit={this.handleAdd}>
                    <input placeholder='Enter new favorite city' />
                    <input type='submit' value='Add city' />
                </form>

                {this.props.favoriteCities.map(city => (
                    <FavoriteCityWeather
                        key={city.name}
                        cityName={city.name}
                        onDelete={this.handleDelete}
                    />
                ))}
            </>
        );
    }
}

const mapStateToProps = ({ favoriteCities }) => {
    return {
        favoriteCities: favoriteCities.cities,
    };
};
const mapDispatchToProps = {
    addCity: addCity,
    deleteCity: deleteCity
};

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);