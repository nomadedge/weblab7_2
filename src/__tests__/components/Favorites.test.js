import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Favorites from '../../components/Favorites';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const existingArray = [
    {
        name: 'Kirov',
        isFetching: true,
        error: null,
        weather: {}
    },
    {
        name: 'Kiyv',
        isFetching: false,
        error: null,
        weather: {
            iconUrl: "http://openweathermap.org/img/w/50n.png",
            description: "mist",
            temperature: "14",
            pressure: 1022,
            humidity: 100,
            windSpeed: 3.6
        }
    }
];

describe('Favorites component', () => {
    test('matches the snapshot on adding', () => {
        const store = mockStore({
            favoriteCities: {
                isAdding: true,
                cities: existingArray
            }
        });

        const tree = renderer.create(
            <Provider store={store}>
                <Favorites />
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});