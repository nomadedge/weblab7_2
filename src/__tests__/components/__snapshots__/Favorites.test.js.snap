// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component matches the snapshot on adding 1`] = `
Array [
  <div
    id="cityAdd"
  >
    <form
      name="cityAdd"
      onSubmit={[Function]}
    >
      <input
        disabled={true}
        id="input"
        placeholder="Enter new favorite city"
      />
      <input
        disabled={true}
        type="submit"
        value="Add city"
      />
    </form>
  </div>,
  <div
    id="favorite"
  >
    <div
      id="weather-container"
    >
      <div
        id="left-block"
      >
        <div
          id="city-name"
        >
          Kirov
        </div>
      </div>
      <div
        id="right-block"
      >
        <div
          id="spinner"
        />
      </div>
    </div>
  </div>,
  <div
    id="favorite"
  >
    <button
      id="delete-button"
      onClick={[Function]}
    >
      Delete city
    </button>
    <div
      id="weather-container"
    >
      <div
        id="left-block"
      >
        <div
          id="city-name"
        >
          Kiyv
        </div>
        <div
          id="icon-temp"
        >
          <img
            alt="Weather icon"
            src="http://openweathermap.org/img/w/50n.png"
          />
          <div
            id="temp"
          >
            14
            Â°C
          </div>
        </div>
      </div>
      <div
        id="right-block"
      >
        <div
          className="info"
        >
          <div>
            Description:
          </div>
          <div
            className="value"
          >
            mist
          </div>
        </div>
        <div
          className="info"
        >
          <div>
            Barometer:
          </div>
          <div
            className="value"
          >
            1022
             hPa
          </div>
        </div>
        <div
          className="info"
        >
          <div>
            Humidity:
          </div>
          <div
            className="value"
          >
            100
             %
          </div>
        </div>
        <div
          className="info"
        >
          <div>
            Wind:
          </div>
          <div
            className="value"
          >
            3.6
             m/s
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
